proto/data.go

```diff
@@ -443,9 +445,15 @@ func MakePriority(r *rand.Rand, userPriority int32) int32 {
 	return math.MaxInt32 - rand.Int31n(math.MaxInt32/userPriority)
 }
 
-// MD5Equal returns whether the specified md5.Size byte arrays are equal.
-func MD5Equal(a, b [md5.Size]byte) bool {
-	for i := 0; i < md5.Size; i++ {
+// TxnIDEqual returns whether the transaction IDs are equal.
+func TxnIDEqual(a, b []byte) bool {
+	if a == nil && b == nil {
```

Comment:
why not scratch everything here up to the for loop before and then at the end return `len(a) == len(b)` instead?